File: BloodFlow360.Domain/Entities/BloodIssue.cs
using BloodFlow360.Domain.Common;

namespace BloodFlow360.Domain.Entities;

public class BloodIssue : BaseEntity
{
    public Guid BloodRequestId { get; set; }

    public BloodRequest BloodRequest { get; set; } = null!;

    public Guid HospitalId { get; set; }

    public Hospital Hospital { get; set; } = null!;

    public string IssueNumber { get; set; } = string.Empty;

    public DateTime IssueDate { get; set; }

    public int TotalUnitsIssued { get; set; }

    public string IssuedBy { get; set; } = string.Empty;

    public string ReceivedBy { get; set; } = string.Empty;

    public string Status { get; set; } = string.Empty;

    public string Remarks { get; set; } = string.Empty;
}

File: BloodFlow360.React/src/types/bloodissue.ts
export interface BloodIssue {
  id: string;
  bloodRequestId: string;
  requestNumber: string;
  bloodGroupName: string;
  hospitalId: string;
  hospitalName: string;
  issueNumber: string;
  issueDate: string;
  totalUnitsIssued: number;
  issuedBy: string;
  receivedBy: string;
  status: string;
  remarks: string;
}

export interface CreateBloodIssue {
  bloodRequestId: string;
  hospitalId: string;
  issueNumber: string;
  issueDate: string;
  totalUnitsIssued: number;
  issuedBy: string;
  receivedBy: string;
  status: string;
  remarks: string;
}

export interface UpdateBloodIssue {
  issueDate: string;
  totalUnitsIssued: number;
  issuedBy: string;
  receivedBy: string;
  status: string;
  remarks: string;
}
